doc/devel/development_setup.rst

```diff
@@ -33,18 +33,31 @@ Retrieving the latest version of the code
 
 Matplotlib is hosted at https://github.com/matplotlib/matplotlib.git.
 
-You can retrieve the latest sources with the command (see
-:ref:`set-up-fork` for more details)::
+After `setting up your own fork <https://docs.github.com/get-started/quickstart/fork-a-repo>`_, you can retrieve the latest
+sources with the command (see for more details)::
 
-    git clone https://github.com/matplotlib/matplotlib.git
+    git clone https://github.com/YOURUSERNAME/matplotlib.git
+    cd matplotlib
+    git remote add upstream https://github.com/matplotlib/matplotlib.git
 
 This will place the sources in a directory :file:`matplotlib` below your
-current working directory.
+current working directory, and tell git all your pull requests should target the
+``upstream`` repository.
```

Comment:
I think this is not correct. The PR target is defined by the repo you forked from. Whether you have an upstream connection does not influence the (default) PR target.

I suggest to just leave off the second sentence. - We maybe don't need to explain every command in detail. Otherwise, you'd have to explain here the full triangular relation between upstream, origin and local (https://www.tomasbeuzen.com/post/git-fork-branch-pull/).

Reply:
How about 

*This will place the sources in a directory :file:`matplotlib` below your current working directory, and set up the ``upstream`` remote to point to the Matplotlib main repository.*

Reply:
Oh nevermind, I had to rebase and saw this was actually ok - no need to mention upstream here at all.

Reply:
I am a bit lost as to the state of things, but do like 

> This will place the sources in a directory :file:matplotlib below your current working directory, and set up the upstream remote to point to the Matplotlib main repository.

I think the source of the problems are that the remote names are not actually semantic, but a bunch of tools have made (incompatible) assumptions about the meanings of names.

Reply:
I added a github documentation link, let me know if it helps clarify the instructions.